### PR TITLE
CSI-5192 uid to name in volume_group_id

### PR DESCRIPTION
**Pull request by ArbelNathan**
now volumeGroupHandle will look like SVC:3;CSI_volumegroup-d2cf8cc7-93b9-4303-b850-079c66525bf0
instead of SVC:3;3

### CODE REVIEW

File: controllers/array_action/array_mediator_svc.py [hunk 1]
```diff
@@ -2020,7 +2021,7 @@ def delete_volume_group(self, volume_group_id):
     def add_volume_to_volume_group(self, volume_group_id, volume_id):
         volume_name = self._get_volume_name_by_wwn(volume_id)
         cli_volume = self._get_cli_volume(volume_name)
-        if cli_volume.volume_group_id and cli_volume.volume_group_id != volume_group_id:
+        if cli_volume.volume_group_name and cli_volume.volume_group_name != volume_group_id:
```
Comment (matancarmeli7) [inline controllers/array_action/array_mediator_svc.py:2024]: why do we do it?

File: controllers/servers/csi/volume_group_server.py [hunk 2]
```diff
@@ -63,15 +63,15 @@ def DeleteVolumeGroup(self, request, _):
...
         array_connection_info = utils.get_array_connection_info_from_secrets(secrets)
 
         with get_agent(array_connection_info, array_type).get_mediator() as array_mediator:
             logger.debug(array_mediator)
 
             try:
-                logger.debug("Deleting volume group {}".format(volume_group_id))
-                array_mediator.delete_volume_group(volume_group_id)
+                logger.debug("Deleting volume group {}".format(volume_group_name))
+                array_mediator.delete_volume_group(volume_group_name)
```
Comment (matancarmeli7) [inline controllers/servers/csi/volume_group_server.py:74]: deleting by id is not safer?

Comment (ArbelNathan) [reply controllers/servers/csi/volume_group_server.py:74]: id is a running number - like "3", it is not unique enough, this way we can be sure we are deleting the right volume. if we had cross storage unique id for volume-group that could be the best.
if we delete by id we can accidentally delete the next volume group that catches the id (on retry, for example)

File: controllers/array_action/array_mediator_svc.py (hunk 1 above)
Comment (ArbelNathan) [reply controllers/array_action/array_mediator_svc.py:2024]: checking if the volume is in vg..

**Review by matancarmeli7 — APPROVED** [review]
LGTM

**Review by ELENAGER — APPROVED** [review]
LGTM

**Pull request merged**